server.js

```diff
@@ -636,31 +640,52 @@ app.post('/submitanswer', function(req, res) {
             return res.status(400).send('Could not find the question');
         }
 
-        logger.log(common.formatString('User {0} attempted to answer question {1} with "{2}"', [userId, questionId, answer]));
+        // check if question is locked for the student
+        questions.isUserLocked(userId, question, function(isLocked, waitTimeMessage, waitTimeinMiliSeconds){
+            if (isLocked){
+                return res.status(423).send(waitTimeMessage);
+            }
 
-        var value = questions.verifyAnswer(question, answer);
-        var points = Math.floor(Math.max(question.minpoints, question.maxpoints/Math.cbrt(question.correctAttemptsCount + 1)));
-        var text = value ? 'correct' : 'incorrect';
-        var status = value ? 200 : 500;
-        var response = {text: text, points: points};
+            logger.log(common.formatString('User {0} attempted to answer question {1} with "{2}"', [userId, questionId, answer]));
 
-        if (req.session.user.type === common.userTypes.ADMIN) {
-            return res.status(status).send(response);
-        }
+            var isCorrect = questions.verifyAnswer(question, answer);
+            var points = 0;
+            var text = 'incorrect';
+            var status = 405;
 
-        users.submitAnswer(userId, questionId, value, points, answer, function(err, result){
-            if(err){
-                logger.error(err);
-                return res.status(500).send(err);
+            if (isCorrect){
```

Comment:
Why can't we check if it is correct before defining all these variables?

Reply:
How do you want to define the variables? they are needed even if it is not correct